classes/controller/FrontController.php

```diff
@@ -477,12 +477,56 @@ public function init()
             $this->displayRestrictedCountryPage();
         }
 
+        if (Tools::isSubmit('live_edit') && !$this->checkLiveEditAccess()) {
+            Tools::redirect('index.php?controller=404');
+        }
+
         $this->iso = $iso;
         $this->context->cart = $cart;
         $this->context->currency = $currency;
     }
 
     /**
+     * Checks if the user can use Live Edit feature
+     *
+     * @return bool
+     */
+    public function checkLiveEditAccess()
+    {
+        if (!Tools::isSubmit('live_edit') || !Tools::getValue('ad') || !Tools::getValue('liveToken')) {
+            return false;
+        }
+
+        if (Tools::getValue('liveToken') != Tools::getAdminToken('AdminModulesPositions'.(int)Tab::getIdFromClassName('AdminModulesPositions').(int)Tools::getValue('id_employee'))) {
+            return false;
+        }
+
+        return is_dir(Tools::safeOutput(_PS_CORE_DIR_.DIRECTORY_SEPARATOR.Tools::getValue('ad')));
```

Comment:
This is not secure. One could inject any path in `ad` parameter.
Please sanitize this by :
- filtering directory separator character
- filtering `.` character
- using `realpath()` to sanitize the result

Reply:
For the second and third points, can you please give me an example or explain the method to call.

Reply:
- If `Tools::getValue('ad')` is empty, throw exception.
- Just remove the `.` character from `Tools::getValue('ad')`

Any hack attempt should result in an invalid path (because we removed the sensible characters).
At the very end, pass the result through `realpath()`.  It returns `false` if the path is invalid.

Reply:
* Test empty value
*********************************************************************************************************************
if (!Tools::isSubmit('live_edit') || !Tools::getValue('ad') || !Tools::getValue('liveToken')) {
            return false;
        }
*********************************************************************************************************************
 * filter path
*********************************************************************************************************************
$adminDir = str_ireplace('.', '', Tools::getValue('ad'));

        $backAccess = realpath(_PS_CORE_DIR_.DIRECTORY_SEPARATOR.$adminDir);
        if ($backAccess === false) {
            return false;
        }

        return is_dir($backAccess);
*********************************************************************************************************************

I hope i understood correctly.<img class="emoji" alt=":smile:" height="20" width="20" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f604.png">